sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -326,36 +329,39 @@ class ContinuousExecution(
    * Mark the specified epoch as committed. All readers must have reported end offsets for the epoch
    * before this is called.
    */
-  def commit(epoch: Long): Unit = {
+  def commit(epoch: Long, epochStats: EpochStats): Unit = {
     updateStatusMessage(s"Committing epoch $epoch")
 
     assert(sources.length == 1, "only one continuous source supported currently")
     assert(offsetLog.get(epoch).isDefined, s"offset for epoch $epoch not reported before commit")
 
-    synchronized {
-      // Record offsets before updating `committedOffsets`
-      recordTriggerOffsets(from = committedOffsets, to = availableOffsets)
-      if (queryExecutionThread.isAlive) {
-        commitLog.add(epoch, CommitMetadata())
-        val offset =
-          sources(0).deserializeOffset(offsetLog.get(epoch).get.offsets(0).get.json)
-        committedOffsets ++= Seq(sources(0) -> offset)
-        sources(0).commit(offset.asInstanceOf[v2.reader.streaming.Offset])
-      } else {
-        return
+    reportTimeTaken("walCommit", epoch) {
+      synchronized {
+        // Record offsets before updating `committedOffsets`
+        recordTriggerOffsets(from = committedOffsets, to = availableOffsets, epoch)
```

Comment:
The `availableOffsets` seems never updated in continuous processing, maybe we should take this into consideration when report metrics.